### PR TITLE
test: Ensure the `custom_wasmer_runtime` tag works

### PR DESCRIPTION
**Pull request by Hywan**
Fixes https://github.com/wasmerio/wasmer-go/issues/191.
Fixes https://github.com/wasmerio/wasmer-go/issues/194.

### CODE REVIEW

**Comment by Hywan** [comment]
bors try

**Comment by Hywan** [comment]
bors try

**Comment by Hywan** [comment]
bors try-

**Comment by Hywan** [comment]
bors try

**Comment by Hywan** [comment]
bors try

**Comment by Hywan** [comment]
bors try

**Comment by Hywan** [comment]
bors try

**Comment by Hywan** [comment]
bors try

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -49,11 +53,31 @@ jobs:
...
+          tar xzvf release.tar.gz
+          cd ..
+          export CGO_CFLAGS="-I$(pwd)/libwasmer/include/"
+          export CGO_LDFLAGS="-L$(pwd)/libwasmer/lib/ -lwasmer"
+          echo $CGO_CFLAGS
+          echo $CGO_LDFLAGS
+          ls $(pwd)/libwasmer/lib/
+          go env
+          cd wasmer
+          go test -v -tags custom_wasmer_runtime
```
Comment (jubianchi) [inline .github/workflows/test.yml:79]:
```suggestion
          just test -tags custom_wasmer_runtime
```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -49,11 +53,31 @@ jobs:
...
+        if: ${{ !matrix.target.libwasmer_archive_name }}
         shell: bash
         run: |
           just test
 
+      - name: Run all the tests (with custom Wasmer runtime)
+        if: matrix.target.libwasmer_archive_name
+        shell: bash
+        run: |
+          libwasmer_url=$(curl -s https://api.github.com/repos/wasmerio/wasmer/releases/latest | jq --raw-output '.assets[] | select(.name == "${{ matrix.target.libwasmer_archive_name }}") | .browser_download_url')
```
Comment (jubianchi) [inline .github/workflows/test.yml:66]:
```suggestion
          libwasmer_url=$(curl -s https://api.github.com/repos/wasmerio/wasmer/releases/latest | jq --raw-output '.assets[] | select(.name == "wasmer-${{ matrix.target.id }}") | .browser_download_url')
```

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -49,11 +53,31 @@ jobs:
...
+          mkdir -p libwasmer
+          cd libwasmer
+          curl -L $libwasmer_url > release.tar.gz
+          tar xzvf release.tar.gz
```
Comment (jubianchi) [inline .github/workflows/test.yml:70]:
```suggestion
          tar xzvf release.tar.gz -C libwasmer
```

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -49,11 +53,31 @@ jobs:
...
         run: |
           just test
 
+      - name: Run all the tests (with custom Wasmer runtime)
+        if: matrix.target.libwasmer_archive_name
+        shell: bash
+        run: |
+          libwasmer_url=$(curl -s https://api.github.com/repos/wasmerio/wasmer/releases/latest | jq --raw-output '.assets[] | select(.name == "${{ matrix.target.libwasmer_archive_name }}") | .browser_download_url')
+          mkdir -p libwasmer
+          cd libwasmer
```
Comment (jubianchi) [inline .github/workflows/test.yml:68]:
```suggestion

```

File: .github/workflows/test.yml [hunk 5]
```diff
@@ -49,11 +53,31 @@ jobs:
...
+          curl -L $libwasmer_url > release.tar.gz
+          tar xzvf release.tar.gz
+          cd ..
```
Comment (jubianchi) [inline .github/workflows/test.yml:71]:
```suggestion

```

File: .github/workflows/test.yml [hunk 6]
```diff
@@ -49,11 +53,31 @@ jobs:
...
+          export CGO_CFLAGS="-I$(pwd)/libwasmer/include/"
+          export CGO_LDFLAGS="-L$(pwd)/libwasmer/lib/ -lwasmer"
```
Comment (jubianchi) [inline .github/workflows/test.yml:73]:
```suggestion
          export CGO_LDFLAGS="-Wl,-rpath,$(pwd)/libwasmer/lib/ -L$(pwd)/libwasmer/lib/ -lwasmer"
```

File: .github/workflows/test.yml (hunk 2 above)
Comment (Hywan) [reply .github/workflows/test.yml:66]: I prefer to not “connect” the matrix target ID with the archive name in this case.

File: .github/workflows/test.yml (hunk 4 above)
Comment (Hywan) [reply .github/workflows/test.yml]: :+1:

File: .github/workflows/test.yml (hunk 3 above)
Comment (Hywan) [reply .github/workflows/test.yml]: :+1:

File: .github/workflows/test.yml (hunk 5 above)
Comment (Hywan) [reply .github/workflows/test.yml]: :+1:

**Comment by Hywan** [comment]
bors try

File: .github/workflows/test.yml (hunk 6 above)
Comment (Hywan) [reply .github/workflows/test.yml]: :+1:

**Comment by Hywan** [comment]
bors try

Normally it works. It fails just because the latest release of Wasmer doesn't include the `wasmer_is_compiler_available` & siblings symbols yet. We can wait on a new release.

**Comment by Hywan** [comment]
bors try

**Comment by Hywan** [comment]
bors try

**Comment by Hywan** [comment]
bors try

**Comment by Hywan** [comment]
bors try

**Pull request merged**